OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -34,7 +34,7 @@ public Attack(Actor self, Target target, bool allowMovement, bool forceAttack)
 
 			this.forceAttack = forceAttack;
 
-			attack = self.Trait<AttackBase>();
+			attack = self.TraitsImplementing<AttackBase>().First(x => !x.IsTraitDisabled);
```

Comment:
Actually, all that needs to be done in other ticks is check if cached trait is disabled and cancel if it is.
```c#
 		public override Activity Tick(Actor self)
 		{
			if (attack == null)
			{
				attack = self.TraitsImplementing<AttackBase>().FirstOrDefault(x => !x.IsTraitDisabled)
				if (attack == null)
					return NextActivity;
			}
			else if (attack.IsTraitDisabled)
				return NextActivity;

 			var ret = InnerTick(self, attack);
 			attack.IsAttacking = ret == this;
 			return ret;
 		}
```